async-dog/index.html

```diff
@@ -18,8 +18,24 @@ <h2>대신 귀여운 강아지를 드립니다.</h2>
     </div>
 
     <script>
-      const url = "https://dog.ceo/api/breeds/image/random";
-      // TODO: url에서 강아지 사진 링크를 가져와 본문의 img 소스를 대체하기
+      function DogApp() {
+        const url = "https://dog.ceo/api/breeds/image/random";
+        const $button = document.querySelector("#fetch-button");
+        const $image = document.querySelector("#dog-image");
+
+        $button.addEventListener("click", event => this.changeDogImage(event));
+
+        this.changeDogImage = (event) => {
+          event.preventDefault();
+          fetch(url)
+                  .then(response => response.json())
+                  .then(item => {
+                    $image.src = item.message;
+                  });
+        }
+      }
+
+      DogApp();
```

Comment:
> 생성자 함수는 반드시 `new` 연산자와 함께 호출해야 합니다. `new`와 함께 호출하면 내부에서 `this`가 암시적으로 만들어지고, 마지막엔 `this`가 반환됩니다.

생성자 함수는 `new` 키워드를 활용하여 초기화해주세요.

```suggestion
      new DogApp();
```